docs/pages/server-access/guides/ec2-discovery.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: Configure Teleport to Automatically Discover EC2 instances
+description: How to configure Teleport to install itself EC2 instances.
+---
+
+Teleport's SSH Service can be configured to automatically discover, and install
+Teleport on newly provisioned EC2 nodes.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- AWS account with EC2 instances and permissions to create and attach IAM policies as well as permission to create System Manager documents.
+- A running Teleport Node. See the [Server Access Getting Started Guide](../getting-started.mdx) for how to add a Node to your Teleport cluster.
+
+## Step 1/4. Create an EC2 invite token
+
+When discovering EC2 nodes Teleport makes use of EC2 invite tokens for
+authenticating joining Nodes.
+
+Create the file `token.yaml`:
+```yaml
+# token.yaml
+---
+kind: token
+version: v2
+metadata:
+  # the token name is not a secret because instances must prove that they are
+  # running in your AWS account to use this token
+  name: iam-invite-token
+  # set a long expiry time, the default for tokens is only 30 minutes
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  # use the minimal set of roles required
+  roles: [Node]
+
+  # set the join method allowed for this token
+  join_method: iam
+
+  allow:
+  # specify the AWS account which nodes may join from
+  - aws_account: "123456789"
+```
+
+Add the token to the Teleport cluster with:
+```
+tctl create -f token.yaml
+```
+
+More documentation on EC2 invite tokens can be found [here](../../management/guides/joining-nodes-aws-iam.mdx)
+
+## Step 2/4. Create an AWS Systems Manager Document
+
+Teleport makes use of AWS Systems Manager documents to execute commands necessary to
+install and start Teleport on discovered EC2 instances.
+
+A document with the following content should be created, named `TeleportDiscoveryInstaller`:
+
+```yaml
+---
+schemaVersion: '2.2'
+description: aws:runShellScript
+parameters:
+  token:
+    type: String
+    description: "(Required) The Teleport invite token to use when joining the cluster."
+  scriptName:
+    type: String
+    description: "(Required) The Teleport instalelr script to use when joining the cluster."
+mainSteps:
+- action: aws:downloadContent
+  name: downloadContent
+  inputs:
+    sourceType: "HTTP"
+    destinationPath: "/tmp/installTeleport.sh"
+    sourceInfo:
+      url: "https://teleport.example.com/webapi/scripts/{{ scriptName }}"
+- action: aws:runShellScript
+  name: runShellScript
+  inputs:
+    timeoutSeconds: '300'
+    runCommand:
+      - /bin/sh /tmp/installTeleport.sh "{{ token }}"
```

Comment:
There's no way for a user to provide this token, right? Teleport always provides it?

If this can be user controlled, we need to be sure to validate it, otherwise someone could provide a token of `&& rm -rf /`.